arch/arm/core/swap_helper.S

```diff
@@ -182,6 +182,10 @@ _thread_irq_disabled:
     bic r3, #1
     orr r3, r0
     msr CONTROL, r3
+#ifdef CONFIG_CPU_CORTEX_M_HAS_ISB
```

Comment:
Should be something like ifdef CONFIG_CPU_CORTEX_M_HAS_UNPRIV

Reply:
Exactly.

Reply:
@galak, now I am thinking whether this is really needed. If the nPRIV bit is not implemented, I believe it is RAZ/WI. So we would not explicitly need the _HAS_UNPRIV option, IMO. Instead, we can keep the current implementation, and just add the __ISB()

Reply:
I agree, now that I look at this, I don't think there is any need for the ifdef at all.